protocols/gossipsub/src/handler.rs

```diff
@@ -409,7 +408,7 @@ impl ConnectionHandler for Handler {
     fn on_behaviour_event(&mut self, message: HandlerIn) {
         match self {
             Handler::Enabled(handler) => match message {
-                HandlerIn::Message(m) => handler.send_queue.push(m),
+                HandlerIn::Message(m) => _ = handler.send_queue.push_back(m), // drops frontmost element if full
```

Comment:
I think it might be useful to log something here but I am undecided on the level. Perhaps `warn!` is justified? If this queue is full repeatedly, we have a problem ..